File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int minl = 400;
        String tm = strs[0];
        for(int i = 0; i < strs.length; i++){
            int idx = 0;
            while(idx < strs[i].length() && idx < minl){
                if(tm.charAt(idx) == strs[i].charAt(idx)) idx++;
                else break;
            }
            minl = Math.min(minl,idx);
        }
        return tm.substring(0,minl);
    }
}

File: 14-longest-common-prefix/longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (a) {
    let n = a.length;
    let m = a[0];
    let cnt = 300;
    for (let i = 0; i < n; i++) {
        let idx = 0;
        let tm = a[i];
        while (idx < tm.length && idx < cnt) {
            if (m.charAt(idx) == tm.charAt(idx)) {
                idx++;
            }
            else break;
        }
        cnt = Math.min(idx, cnt);
    }
    return m.substring(0, cnt);
};